src/GhostGeneration.jl

```diff
@@ -0,0 +1,128 @@
+# defines and stores relevant variables to ghost creation and area calculation
+struct GhostedArea
+    resolution::Int
+    n_ghosts::Int
+    rapidity_step::Float64
+    phi_step::Float64
+    ghost_pt::Float64
+
+    # constructor
+    # default value of ghost_pt set to 1.0e-45 as -45 is the smallest magnitude that doesn't result in the value defaulting to 0
+    function GhostedArea(resolution::Int)
+        new(resolution,
+            resolution * resolution,
+            Float64(10.0 / resolution),
+            Float64(6.28 / resolution),
+            1.0e-45)
+    end
+end
+
+# generates and adds ghosts to an event (a set of PseudoJets)
+function add_ghosts!(event::Vector{PseudoJet}, gen::GhostedArea)
+    # Set aside memory for output
+    n_original = length(event)
+    resize!(event, n_original + gen.n_ghosts)
+
+    # Generate random values
+    rand_vals = rand(Float64, 2 * gen.n_ghosts) .- 0.5f0
+
+    # loop without array bounds checking in order to maximize time efficiency
+    @inbounds for k in 0:(gen.n_ghosts - 1)
+        i = k ÷ gen.resolution
+        # modular division means j cycles through 0-99 100 times (acts like an inner loop)
+        j = k % gen.resolution
+        # index to iterate through the random values
+        index = 2k + 1
+
+        rap = (i + rand_vals[index]) * gen.rapidity_step
+        phi = (j + rand_vals[index + 1]) * gen.phi_step
+
+        # k + 1 is due to indexing beginning at 1 in julia
+        event[n_original + k + 1] = PseudoJet(pt = gen.ghost_pt, rap = rap, phi = phi, _pure_ghost = true)
+    end
+
+    return event
+end
+
+# calculate the number of ghosts in a jet
+function ghosts_in_jet(cluster_seq::ClusterSequence, jet::PseudoJet)
+
+    # Get the constituents of the first jet
+    jet_constituents = JetReconstruction.constituents(jet, cluster_seq)
+
+    # determine the number of ghosts in the jet by checking each constituent's _pure_ghost field
+    num_ghosts = 0
+    for c in jet_constituents
+        if c._pure_ghost
+            num_ghost += 1
+        end
+    end
+
+    # return the number of ghosts
+    return num_ghosts
+end
+
+# calculate the area of a jet
+function GhostedAreaCalculation(cluster_seq::ClusterSequence, jet::PseudoJet, resolution::Int)
+    # the total area is the rapidity range (-5,5) times the phi range (0, 2pi)
+    total_area = 10.0 * 6.28
```

Comment:
Can use `2π` here. In principle the total area depends on the rapidity & azimuthal range: instead of using 10, this should use e.g. `abs(rap_max-rap_min)`.